src/components/Trade/hooks/useSwapper/useSwapper.tsx

```diff
@@ -151,6 +166,91 @@ export const useSwapper = () => {
     ],
   )
 
+  const checkApprovalNeeded = useCallback(async () => {
+    const adapterManager = getChainAdapterManager()
+    const adapter = adapterManager.get(sellAsset.chainId)
+
+    if (!adapter) throw Error(`no chain adapter found for chain Id: ${sellAsset.chainId}`)
+    if (!wallet) throw new Error('no wallet available')
+    if (!activeQuote) throw new Error('no activeQuote available')
+
+    // No approval needed for selling a fee asset
+    if (sellAsset.assetId === adapter.getFeeAssetId()) {
+      return false
+    }
+
+    const from = await adapter.getAddress({
+      wallet,
+      accountNumber: activeQuote.accountNumber,
+    })
+
+    const { assetReference } = fromAssetId(sellAsset.assetId)
+    const web3 = getWeb3InstanceByChainId(sellAsset.chainId)
+
+    const allowanceOnChainCryptoBaseUnit = await getERC20Allowance({
+      web3,
+      erc20AllowanceAbi,
+      address: assetReference,
+      spender: activeQuote.allowanceContract,
+      from,
+    })
+
+    return bn(allowanceOnChainCryptoBaseUnit).lt(activeQuote.sellAmountBeforeFeesCryptoBaseUnit)
+  }, [activeQuote, sellAsset.assetId, sellAsset.chainId, wallet])
+
+  const getApprovalTxData = useCallback(
```

Comment:
Note, this doesn't return EVM calldata which one would expect from the naming, but `buildCustomTxInput` and `networkFeeCryptoBaseUnit`, though I don't have a better naming in mind, just something to keep in mind if we think of a better naming 🙏🏽